aquacomputer_d5next.c

```diff
@@ -124,16 +208,420 @@ struct d5next_data {
+        &sensor_dev_attr_pwm1_auto_point12_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm1_auto_point13_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm1_auto_point14_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm1_auto_point15_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm1_auto_point16_temp.dev_attr.attr,
+        
+        &sensor_dev_attr_pwm1_auto_start_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm1_min.dev_attr.attr,
+        &sensor_dev_attr_pwm1_max.dev_attr.attr,
+
+        &sensor_dev_attr_pwm2_auto_point1_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point2_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point3_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point4_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point5_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point6_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point7_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point8_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point9_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point10_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point11_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point12_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point13_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point14_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point15_pwm.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point16_pwm.dev_attr.attr,
+
+        &sensor_dev_attr_pwm2_auto_point1_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point2_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point3_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point4_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point5_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point6_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point7_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point8_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point9_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point10_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point11_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point12_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point13_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point14_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point15_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_auto_point16_temp.dev_attr.attr,
+        
+        &sensor_dev_attr_pwm2_auto_start_temp.dev_attr.attr,
+        &sensor_dev_attr_pwm2_min.dev_attr.attr,
+        &sensor_dev_attr_pwm2_max.dev_attr.attr,
+        NULL,
+};
+
+static umode_t d5next_auto_pwm_is_visible(struct kobject *kobj, struct attribute *attr, int index)
+{
+//    struct device *dev = kobj_to_dev(kobj);
+//    struct d5next_data *data = dev_get_drvdata(dev);
+    //todo don't show auto settings unless curve mode is enabled?
```

Comment:
Following the example of it87 and nct6775, if curve support exists it is always shown. This is more for hiding features the current HW doesn't support.

Otherwise, you would be unable to configure the fan curve before enabling it, which would be bad.

Reply:
Oh, so auto refers to the curves? If so, don't hide it as you say

Reply:
That's correct. Specifically, it refers to fan control mode `2`. Mode `1` is PID, which has a different suite of settings that I haven't exposed yet. (not sure if I want to. Hwmon doesn't seem to have an idiomatic way to expose PID tuning params :P)

Reply:
Yeah, if it can't use an existing interface it's best left to userspace. Similar story with RGB.